text/0000-location.md

```diff
@@ -0,0 +1,440 @@
+organization in the location’s owner field. A setting will turn off deletion 
+entirely, with the intent that it could be enabled/disabled by a Grid 
+administrator. Disabling delete is useful because external systems may have 
+references to these locations and deleting them could leave dangling references.
+
+
+## GLN Validation
+[gln validation]: #glnvalidation
+
+GLN validation can reduce the opportunity for invalid data to enter the system.
+At a minimum, when GLNs are entered in clients and later processed in smart
+contracts, they will need to conform to the GLN format and the check digit
+should be correct. See [check digit calculator](https://www.gs1us.org/tools/check-digit-calculator "check digit calculator").
+
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## State
+[state]: #state
+
+### Location Representation
+
+The primary object stored in state is “Location”, which consists of a 
+location_id (GS1 GLN), an owner (org_id compatible with Pike), and a list of 
+property name/value pairs. The properties available are defined by the Grid 
+Location Schema Transaction Family and are restricted to the fields and rules of
+the GS1 Location Property Schema. Transactions which are responsible for setting
+location state values must ensure that the properties conform with the 
+requirements of the GS1 Location Property Schema.
+
+    message Location { 
+        enum LocationNamespace { 
+            UNSET_NAMESPACE = 0; 
+            GS1 = 1; 
+        }
+
+        // What type of location is this (GS1)
+        LocationNamespace location_namespace = 1; 
+
+        // location_id is the GLN for GS1-namespace locations
+        string location_id = 2;
+
+        // Who owns this location (pike organization id)
+        string owner = 3;
+
+        // Properties about this location
+        repeated PropertyValue properties = 4; 
+    }
+
+The GS1 GLN is an identifier (location_id) used to identify entities or 
+locations. A GLN is the data transmitted from a data carrier (barcode, rfid, 
+etc.) scan and is made up of a company prefix, location reference, and check 
+digit.
+
+### Referencing Locations
+
+Locations are uniquely referenced by their location_id and location_namespace.
+For GS1, locations are referenced by the GLN identifier.
+
+**Note: For the remainder of this document, “address” will refer to a 
+Grid/Transact address as opposed to a Street Address.**
+
+### Location Addressing in the Merkle-Radix State System
+
+In order to uniquely locate GS1 locations in the Merkle-Radix state system, an 
+address must be constructed which identifies the storage location of the 
+Location representation.
+
+All Grid addresses are prefixed by the 6-hex-character namespace prefix “621dee”.
+Locations are further prefixed under the Grid namespace with reserved 
+enumerations of “04” (“00”, “01”, “02” and “03” being reserved for other 
```

Comment:
Should the definition of these reservations be added here for future us to be able to reference?

Reply:
Sure thing - each reservation is now defined within the RFC.